src/context.rs

```diff
@@ -65,31 +68,45 @@ extern "C" {
   );
 }
 
+#[derive(Default)]
+pub struct ContextOptions<'s> {
+  /// An optional object template from which the global object for the newly created context will
+  /// be created.
+  pub global_template: Option<Local<'s, ObjectTemplate>>,
+  /// An optional global object to be reused for the newly created context. This global object
+  /// must have been created by a previous call to Context::New with the same global template.
+  /// The state of the global object will be completely reset and only object identify will
+  /// remain.
+  pub global_object: Option<Local<'s, Value>>,
+  /// An optional microtask queue used to manage the microtasks created in this context. If not
+  /// set the per-isolate default microtask queue would be used.
+  pub microtask_queue: Option<*mut MicrotaskQueue>,
```

Comment:
I'm planning to make this a `shared_ptr` upstream so that we don't have this ugly raw pointer.